src/codemagic/cli/argument/argument_parser_builder.py

```diff
@@ -123,16 +124,42 @@ def _get_custom_argument_group(self, group_name) -> ArgumentGroup:
             self._custom_arguments_groups[group_name] = argument_group
         return argument_group
 
+    def _get_custom_mutually_exclusive_group(
+        self,
+        mutually_exclusive_group: MutuallyExclusiveGroup,
+    ) -> ArgumentGroup:
+        try:
+            argument_group = self._custom_arguments_groups[mutually_exclusive_group.group_name]
+        except KeyError:
+            group_description = Colors.UNDERLINE(
+                f"Following arguments are mutually exclusive for {Colors.BOLD(self._cli_action.action_name)}",
+            )
+            group = self._action_parser.add_argument_group(group_description)
+            argument_group = group.add_mutually_exclusive_group(required=mutually_exclusive_group.required)
+            self._custom_arguments_groups[mutually_exclusive_group.group_name] = argument_group
+        return argument_group
```

Comment:
Related to [this outdated comment](https://github.com/codemagic-ci-cd/cli-tools/pull/422#discussion_r1704404774).

I read from the docs that it is just another possibility to isolate mutually exclusive arguments into separate groups with a title a description, but I fail to see that this approach is or should be favored.

If all mutually exclusive groups are indeed going to be explicitly separated, then we need proper titles and descriptions for them, which would properly explain their usages, and also make them compatible with the already supported "custom argument groups".

Given that, at least something like as follows is necessary:

```python
    def _get_mutually_exclusive_group(self, group: MutuallyExclusiveGroup) -> ArgumentGroup:
        try:
            mutually_exclusive_argument_group = self._custom_arguments_groups[group.name]
        except KeyError:
            if group.required:
                title_prefix, description_prefix, description_verb = "Required", "Exactly", "must"
            else:
                title_prefix, description_prefix, description_verb = "Optional", "Only", "can"

            title = (
                f"{title_prefix} mutually exclusive arguments "
                f"for command {self._cli_action.action_name} to {Colors.BOLD(group.name)}"
            )
            argument_group = self._action_parser.add_argument_group(
                Colors.UNDERLINE(title),
                f"{description_prefix} one of those options {description_verb} be selected",
            )
            mutually_exclusive_argument_group = argument_group.add_mutually_exclusive_group(required=group.required)
            self._custom_arguments_groups[group.name] = mutually_exclusive_argument_group
        return mutually_exclusive_argument_group
```

Perhaps add methods to generate title and description to `MutuallyExclusiveGroup`?